File: android/Detonator/src/main/java/com/iconshot/detonator/request/InputSetValueRequest.java
package com.iconshot.detonator.request;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.element.InputElement;
import com.iconshot.detonator.renderer.Edge;

public class InputSetValueRequest extends Request<String> {
    public InputSetValueRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        String value = decodeData(String.class);

        Edge edge = getComponentEdge();

        InputElement element = (InputElement) edge.children.get(0).element;

        element.setValue(value);

        end();
    }
}

File: ios/Detonator/Sources/Detonator/Request/InputSetValueRequest.swift
class InputSetValueRequest: Request {
    override public func run() {
        let value: String! = decodeData()
        
        let edge = getComponentEdge()!
        
        let element = edge.children[0].element as! InputElement
        
        element.setValue(value: value)
        
        end()
    }
}
